go/tools/gazelle/packages/package.go

```diff
@@ -0,0 +1,118 @@
+/* Copyright 2017 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package packages
+
+import "strings"
+
+// Package contains metadata about a Go package extracted from a directory.
+// It fills a similar role to go/build.Package, but it separates files by
+// target instead of by type, and it supports multiple platforms.
+type Package struct {
+	Dir  string
```

Comment:
The second one: planning to return `[]Package` eventually.

I could see taking `Dir` out at that point and passing it in as a separate parameter to `rules.Generator.Generate`, but I'm not sure I want to do that refactoring yet.